packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +22,222 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
 ///         challenged within the challenge period. By using stored anchor states, new FaultDisputeGame instances can
 ///         be initialized with a more recent starting state which reduces the amount of required offchain computation.
 contract AnchorStateRegistry is Initializable, ISemver {
-    /// @notice Describes an initial anchor state for a game type.
-    struct StartingAnchorRoot {
-        GameType gameType;
-        OutputRoot outputRoot;
-    }
+    error InvalidGame();
+    error AnchorGameIsNewer(uint256 anchorGameL2BlockNumber, uint256 candidateGameL2BlockNumber);
+
+    event AnchorGameSet(IDisputeGame newAnchorGame);
+    event RespectedGameTypeSet(GameType gameType);
+    event DisputeGameBlacklisted(IDisputeGame game);
+    event GameRetirementTimestampSet(uint64 timestamp);
 
     /// @notice Semantic version.
+    // TODO: update?
     /// @custom:semver 2.0.1-beta.6
     string public constant version = "2.0.1-beta.6";
 
     /// @notice DisputeGameFactory address.
-    IDisputeGameFactory internal immutable DISPUTE_GAME_FACTORY;
+    IDisputeGameFactory public disputeGameFactory;
+
+    /// @notice The delay between when a dispute game is resolved and when it can be considered finalized.
+    uint256 public disputeGameFinalityDelaySeconds;
 
-    /// @notice Returns the anchor state for the given game type.
-    mapping(GameType => OutputRoot) public anchors;
+    // TODO: storage slot issue
```

Comment:
both here and in portal, we are upgrading a proxy's implementation, right? assuming i should include gaps for storage sots.

Reply:
Yeah good point. Including gaps would be best practice, let's make sure the storage layout lines up.